crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -296,19 +301,27 @@ impl<'a> Compressor<'a> {
                         let bool_literal = self.hir.boolean_literal(unary_expr.span, !boolean);
                         return Some(self.hir.literal_boolean_expression(bool_literal));
                     }
-                }
+                    Expression::BigintLiteral(big_int_literal) => {
+                        if big_int_literal.value.is_zero() {
```

Comment:
Maybe we don't need this check. 🤔 `!0n` can be folded to `true`, and then will be compressed to `!0` in [`compress_boolean`](https://github.com/Boshen/oxc/blob/b8f5e3ad929a041c6b980b139bebe86fb398729a/crates/oxc_minifier/src/compressor/mod.rs#L286). Which is shorter.